cms/jinja2/templates/base.html

```diff
@@ -35,6 +35,59 @@
     {%- endwith -%}
 {%- endset -%}
 
+{% set navigation_settings = settings.navigation.NavigationSettings %}
+
+{% set keyLinksList = [] %}
+
+{% if navigation_settings.main_menu and navigation_settings.main_menu.highlights %}
+    {% for highlight in navigation_settings.main_menu.highlights %}
+        {% if highlight.value.page %}
```

Comment:
```suggestion
        {% if highlight.value.page and highlight.value.page.live %}
```

so we don't inadvertently display a non-published page.

this would need to be reflected below too (for sections)

Reply:
Done!